addon/index.js

```diff
@@ -188,6 +280,26 @@ export class EmberChangeset extends BufferedChangeset {
 
     return this;
   }
+
+  get(key) {
+    safeGet(this[VIRTUAL_MIRROR], key); //consume tag for native tracked dependencies
+    return super.get(...arguments);
+  }
+
+  addObserver(key, callback) {
+    if (!deepHasOwnProperty(this[VIRTUAL_MIRROR], key)) {
+      this.setDeep(this[VIRTUAL_MIRROR], key);
+    }
+
+    deepAddObserver(this[VIRTUAL_MIRROR], key, callback);
+  }
```

Comment:
Manually adding observers in general is not that common, I needed them for a very specific use case that I'm not entirely sure if it could be solved by native getters and this mirror approach, still having a way to add them gives more flexibility and an actual way to do it, which doesn't exist today, also... adding them to the `content` have it's own set of inconveniences